### DISCUSSION TITLE
Anycable w/Rails / puma-dev w/SSL

### DISCUSSION THREAD

**Discussion by braindeaf** [anycable/anycable#225] [General]
Hi there,

I wonder if I can get some idea on how to progress, we've been running our applications locally with SSL using puma-dev for sometime now. It make sense to us to use https://blah.company.test even locally as we run multiple applications sharing the same cookie domain, etc. So we mirror our staging / production environment. Anyway. This means when it comes to running anycable we have say

```
http://blah.company.test -> ws://blah.company.test:8080/cable
```
so I'd expect to have something like.

```
https://blah.company.test -> wss://blah.company.test:8081/cable
```

However, I get

```
WebSocket connection to 'wss://blah.company.test:8081/cable' failed: 
```

Ok I think that's because I need anycable-go to act as an SSL endpoint too. So I'm running.

```
bin/anycable-go --port=8081 --headers=cookie,origin -debug --ssl_cert=~/Library/Application Support/io.puma.dev/cert.pem --ssl_key=~/Library/Application Support/io.puma.dev/key.pem
```

Still WebSocket connection is failing. I'm not sure how to proceed with this. In production this is going to be fine because it will be behind Nginx so I can cope with that but in development it currently doesn't seem possible. Any thoughts?

Many thanks

RobL

**Comment by palkan** [comment]
With puma-dev, you don't need to run AnyCable-Go with SSL enabled (that's the responsibility of the proxy), i.e. the scheme must be as follows:

```sh
https://blah-cable.company.test -> ws://localhost:8081/cable
```

(or `echo "8081" > ~/.puma-dev/blah-cable)

**Reply by braindeaf** [reply]
Thank you that's just what I needed. I've used a different hostname and puma-dev proxies to the right port. Yay. Very much appreciated.
